### ISSUE TITLE
[ENH] File Manager Integration

### ISSUE THREAD

**Issue by thabaum** [ADefWebserver/ADefWebserver.Module.HtmlTextV2#8]
The Oqtane Framework File Manager will need to be integrated, enhanced and tested for inserting document/media/image files.

**Comment by ADefWebserver** [comment]
Can't we just use the existing endpoint that the current core HTML module uses? Meaning, do *not* implement our own API endpoint. We could even disable the Radzen HTML upload popup if needed. This way if the core makes a security fix on the upload API it would automatically apply to this module too.

**Comment by thabaum** [comment]
I agree with utilizing the API from the core as much as we can, and enhance it as needed to meet our needs with some enhancement requests.

I am curious if we can make a better file manager that has one that looks like the upload picker while we are poking around here... while using the current API.  I mean that is the point of Oqtane is to be able to write apps like this.

I do feel simple solution to view files should work to retrieve data securely that a user is allowed.  We need to use best practices here.  I am just thinking outside the box a little as it may bring on new enhancements to the framework itself to allow all of this to happen.

**Comment by thabaum** [comment]
Possible Idea:

![image](https://github.com/ADefWebserver/ADefWebserver.Module.HtmlTextV2/assets/13577556/3401a7c7-e496-493d-a584-7474429a292c)

integrate the file manager as a component here put the configured file manager component here: 
![image](https://github.com/ADefWebserver/ADefWebserver.Module.HtmlTextV2/assets/13577556/79795e9b-9ee6-4b4f-a205-f78efa25aaab)
Add:
![image](https://github.com/ADefWebserver/ADefWebserver.Module.HtmlTextV2/assets/13577556/b2ff3dce-24e0-456b-a64e-dc353f84767e)

**Comment by ADefWebserver** [comment]
All great ideas and I agree with you I just want to get a version out, that works, that just meets the minimal requirements :)

**Comment by thabaum** [comment]
I hope we can release with a future proof version best we can taking in all ideas as they come.  Again this will be installable but needs to give a real reason to support or becomes uninstallable.  If we can't make this work, who can?!

**Comment by thabaum** [comment]
As for this POC i just want to see it work with saving content and switching between design and source modes as expected in all render modes that can be chosen today in Oqtane.  The file manager can stay same but I feel it can possibly be enhanced further with not a lot of effort.

I am grinding my way there, might have something sooner than later I am hoping. 🚀 🌔

**Comment by thabaum** [comment]
When I say use the file manager I am saying put the UI component in that location.  It is already created, use it and replace the other one that just has logic for an upload.  Either way I am open to for now for sure ;)

**Comment by ADefWebserver** [comment]
> When I say use the file manager I am saying put the UI component in that location. It is already created, use it and replace the other one that just has logic for an upload.

Ok sounds good 👍🏽

**Comment by thabaum** [comment]
![image](https://github.com/ADefWebserver/ADefWebserver.Module.HtmlTextV2/assets/13577556/904a2295-c9bf-4c2e-8474-e63a490f8c7c)

This is what I am proposing we think about doctoring up with the file manager.

**Comment by ADefWebserver** [comment]
It looks great, but, I think it is a lot of work to deal with all the "issues" (security, features, ect.)
